cockpit/src/main/java/com/polidea/cockpit/manager/CallbackNotifier.kt

```diff
@@ -1,27 +1,53 @@
 package com.polidea.cockpit.manager
 
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.OnLifecycleEvent
 import com.polidea.cockpit.event.ActionRequestCallback
 
 internal class CallbackNotifier {
 
-    private val callbacks: MutableMap<String, MutableSet<ActionRequestCallback>> = mutableMapOf()
+    private val callbacks: MutableMap<String, MutableSet<LifecycleAwareCallback>> = mutableMapOf()
 
     fun add(paramName: String, callback: ActionRequestCallback) {
         val paramListeners = getCallbacks(paramName)
-        paramListeners.add(callback)
+        paramListeners.add(LifecycleAwareCallback(null, callback, paramName, this))
+        callbacks[paramName] = paramListeners
+    }
+
+    fun add(lifecycleOwner: LifecycleOwner, paramName: String, callback: ActionRequestCallback) {
+        val paramListeners = getCallbacks(paramName)
+        paramListeners.add(LifecycleAwareCallback(lifecycleOwner, callback, paramName, this))
         callbacks[paramName] = paramListeners
     }
 
     fun remove(paramName: String, callback: ActionRequestCallback) {
-        callbacks[paramName]?.remove(callback)
+        val toRemove = callbacks[paramName]?.find { it.callback == callback }
+        callbacks[paramName]?.remove(toRemove)
     }
 
     fun requestAction(paramName: String) {
         val paramListeners = getCallbacks(paramName)
         paramListeners.forEach {
-            it.onActionRequested()
+            it.callback.onActionRequested()
         }
     }
 
-    private fun getCallbacks(name: String): MutableSet<ActionRequestCallback> = callbacks[name] ?: mutableSetOf()
+    private fun getCallbacks(name: String): MutableSet<LifecycleAwareCallback> = callbacks[name] ?: mutableSetOf()
+}
+
+private class LifecycleAwareCallback(val lifecycleOwner: LifecycleOwner?,
+                                     val callback: ActionRequestCallback,
+                                     val paramName: String,
+                                     val notifier: CallbackNotifier) : LifecycleObserver {
```

Comment:
I think you could put whole `LifecycleAwareCallback` class inside `CallbackNotifier` and that would allow `LifecycleAwareCallback` class to access `callbacks` map directly. Without passing `CallbackNotifier` instance to every `LifecycleAwareCallback`.